### DISCUSSION TITLE
sympy.physics.mechanics test function

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#27144] [Q&A]
What would be the practical limit on the running time of a test function?
Reason for my question. If I want to test, that nonlinear motion constraints are kept, one way to do it is to integrate the simulation numerically and show that the nonlinear motion constraints are kept well, say | complex motion constraints| < $10^{-5}$.
**Thanks!**

**Comment by moorepants** [comment]
Unit tests, in general, should be very fast because we run hundreds of them over and over again. There are some tests marked as "slow" and if you search on this repository you'll find whatever the time cutoff is for a "slow" test. Ideally we don't add any slow tests unless it is the only way to test what is needed. You can always open a PR and try to get something in and the reviewers will tell you if it is too slow.

**Reply by Peter230655** [reply]
**Thanks!**
1.
So, it would be better to compare fr, frstar I get with a manually worked out example, even if it is very simple?
(There is one in _Forces associated with non-linear, non-holonomic constraint equations,_ by Roithmayr and Hodges, in International Journal of Non_linear Mechanics, 2009)
2.
I think, these Github test will also tell me, if my code is much slower than the code it replaces?

**Reply by moorepants** [reply]
>  So, it would be better to compare fr, frstar I get with a manually worked out example, even if it is very simple?

Yes, that is likely better.

>  I think, these Github test will also tell me, if my code is much slower than the code it replaces?

No, it will not. You can test speed of changes locally on your computer.

**Reply by Peter230655** [reply]
1.
Clear thanks!

2.
I have been doing this. Then I simply misunderstood some message from Github.

Thanks!

**Reply by moorepants** [reply]
There is a benchmarking message from Github on PRs and this does check timing of things, maybe that is what you are thinking of.

**Reply by Peter230655** [reply]
Yes, this is what I thought of.

Question: 
I have never published a paer in my live - and likely never will.
Was my quotation of this article as I wrote above correct? (I believe a test function shoulf be correct in _every_ aspect)

**Thanks!**
